core/src/main/scala/com/yahoo/maha/core/FilterOperation.scala

```diff
@@ -943,8 +1023,13 @@ object FilterSql {
     filter match {
       case PushDownFilter(f) => 
         renderFilter(f, aliasToNameMapFull, columnsByNameMap, engine, literalMapper, expandedExpression)
+      case FieldEqualityFilter(f,g, _, _) =>
+        require(aliasToNameMapFull.contains(g))
```

Comment:
We don't want to do validation like this at render time.  We should do this validation before we do all the hard work to get this far.  Ideally in request model generation.